File: Aeds 2/TP1/PalindromoRecursivo.c
#include <stdio.h>
#include <string.h>

int isPalindromo(char text[1000], int i){
  if((strlen(text)/2)==i){
    return 1;
  }else if (text[i]==text[strlen(text)-1-i]){
    return isPalindromo(text,i+1);
  }else{
    return 0;
  }
}

int main(void) {
  char text[1000];
  scanf(" %[^\n]",text);

  while(strcmp(text,"FIM")!=0){
    int i=0;
    if(isPalindromo(text,i)){
      printf("SIM\n");
    }else{
      printf("NAO\n");
    }
    scanf(" %[^\n]",text);
  }
  
  
  return 0;
}

File: Aeds 2/TP1/PalindromoRecursivo.java


class PalindromoRecursivo{
    static boolean isPalindromo(String text,int i){
        if(i==text.length()/2){
            return true;
        }
        if(text.charAt(i)==text.charAt(text.length()-1-i)){
            return isPalindromo(text, i+1);
        }else{
            return false;
        }
    }
    public static void main(String[] args) {
        String text;
        text=MyIO.readLine();
        while(text.equals("FIM")!=true){
            int i=0;
            if(isPalindromo(text,i)){
                MyIO.println("SIM");
            }else{
                MyIO.println("NAO");
            }
            text=MyIO.readLine();
        }
    }
}